butterfly/ImageLayer/Datasource.py

```diff
@@ -3,3 +3,7 @@ class Datasource(object):
     @staticmethod
     def load_tile(query):
         return None
+
+    @staticmethod
+    def preload_source(query):
+        return None
```

Comment:
All Datasource methods will need to overwrite `preload_source` to take a `TileQuery` and return a list of 
- the blocksize as a `np array` of shape (3,) 
- the datatype as a string of the form 'uint32'